src/covid_model_seiir_pipeline/marshall.py

```diff
@@ -0,0 +1,442 @@
+"""
+Code for marshalling data (currently only DataFrames) to/from disk.
+
+This module holds 2 key concepts meant to be used externally - Keys and Marshall classes.
+
+Keys are used to identify data types and how they are stored. This is by
+embedding related information with them (draw_id for regression and location_id
+for forecast).
+
+Marshall classes come in several flavors named after the related storage backend.
+Marshall classes should created via Marshall.from_paths(paths_cls) until Paths
+are fully deprecated and removed.
+"""
+from collections import defaultdict
+from contextlib import contextmanager
+import io
+from pathlib import Path
+import typing
+import zipfile
+
+import h5py
+import numpy
+import pandas
+
+from covid_model_seiir_pipeline.paths import (
+    Paths,
+    DRAW_FILE_TEMPLATE,
+)
+from covid_shared.shell_tools import mkdir
+
+
+class DataTypes:
+    """
+    Enumerations of data types as understood by the Marshall interface.
+    """
+    beta_scales = "beta_scales"
+    coefficient = "coefficients"
+    components = "components"
+    date = "dates"
+    fit_beta = "betas"  # TODO: no longer needed - fit is not a stage
+    parameter = "parameters"
+    regression_beta = "beta"
+
+    # types which serialize to a DataFrame
+    DataFrame_types = frozenset([
+        fit_beta,
+        parameter,
+        date,
+        coefficient,
+        regression_beta,
+        beta_scales,
+        components,
+    ])
+
+
+class Keys:
+    """
+    Identifies data payload types and stores related information.
+
+    Callers should never use init and instead use one of the many factory
+    methods which embed important related information to the variable
+    draw_id/location_id argument.
+    """
+    def __init__(self, data_type, template, **key_args):
+        self.data_type = data_type
+        self.template = template
+        self.key_args = key_args
+
+    # factories related to regression
+    @classmethod
+    def coefficient(cls, draw_id):
+        return cls(DataTypes.coefficient, DRAW_FILE_TEMPLATE, draw_id=draw_id)
```

Comment:
DRAW_FILE_TEMPLATE is `draw_{draw_id}.csv`

This gets used in conjunction with `Keys.key` (a property) to generate expected file names